src/api/services/userService.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { NextFunction, Response } from 'express';
+import { Knex } from 'knex';
+
+import { TestConfigure } from '../../database/TestSelfServeDatabase';
+import {
+  createParticipant,
+  createResponseObject,
+  createUser,
+  createUserParticipantRequest,
+  getParticipantIdsOfUser,
+} from '../../testHelpers/apiTestHelpers';
+import { verifyAndEnrichUser } from '../middleware/usersMiddleware';
+import * as kcUsersService from './kcUsersService';
+import { UserService } from './userService';
+
+describe('User Service Tests', () => {
+  let knex: Knex;
+  let next: NextFunction;
+  let res: Response;
+
+  beforeEach(async () => {
+    knex = await TestConfigure();
+    next = jest.fn();
+    ({ res } = createResponseObject());
+    jest.spyOn(kcUsersService, 'removeApiParticipantMemberRole').mockResolvedValue();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('User removal', () => {
+    describe('User still belongs to a participant', () => {
+      it('removes the user from the correct participant', async () => {
+        const targetParticipant = await createParticipant(knex, {});
+        const anotherParticipant = await createParticipant(knex, {});
+        const user = await createUser({
+          participantToRoles: [
+            { participantId: targetParticipant.id },
+            { participantId: anotherParticipant.id },
+          ],
+        });
+        const request = createUserParticipantRequest(user.email, targetParticipant, user.id);
+        await verifyAndEnrichUser(request, res, next);
```

Comment:
We are calling the middleware here to enrich the request with more information. i.e. without this, the `{ user }` in the request is undefined. Don't love this but not sure of another approach that is better.